Architecture.md

```diff
@@ -71,6 +71,79 @@
+
+The advantage of this solution is that it is lightweight and give us full control over the authentication and authorization of the clients. It also allows us to use the same authentication and authorization system as the rest of the application, which is easier to maintain.
+
+The git server is written in golang, for its simplicity and its performance. It is also well documented, and has a lot of libraries to help us build the server, unlike rust which was the original choice. The libraries used are [ssh made by gliderlabs](https://github.com/gliderlabs/ssh) which is used to create git clients and servers. It is also used by [gitea](https://about.gitea.com) to create their git server.
+
+The authentication and authorization process is the following:
+
+- The client connects to the git server using ssh
+- The git server checks if the client is authenticated using the ssh key provided by the client by matching it against the ssh keys stored in the database
+    - If the client is not authenticated, the connection is closed
+- The git server checks if the client is authorized to push to the repository by checking if the client is the owner of the repository
+    - If the client is not authorized, the connection is closed
+- The git server executes the command `git-receive-pack` which is used to push to a repository
+
+##### Building images
+
+Once the code is finally pushed to the repository, it needs to be built in order to be deployed. Since the strategy we chose is [Docker](https://www.docker.com/), the code needs to be built into a docker image.
+
+In that essence, since the client code can be written in any language, we need to be able to build the code in any language. To do so, we use [buildpacks](https://buildpacks.io/), which is a tool to build code in any language into a docker image. It is used by [heroku](https://www.heroku.com/) to build their applications.
+
+How can we trigger the build of the image ? For a first version, we chose to use [git hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) to trigger a bash script which will build the image on the host machine. This is not scalable, as the build is done on the host machine. It also poses questions about security risks since our ssh server needs to be able to access the docker daemon on the host machine. But it is a good first step to test the concept.
+
+Some builders might not be able to build the code without configuration, hence, we need to be able to configure the buildpacks. To do so, the client can store a file called `buildpacks.json` in the `paastech` directory which will allow the user to configure the buildpacks. The file is a json file, which contains the configuration of the buildpacks. It is then used by the git hook to configure buildpacks. Here's the structure of the file:
+
+```json
```

Comment:
Change from JSON to TS, as this is not the JSON syntax

Reply:
its not been converted to yaml as per specifications from the cli @BioTheWolff